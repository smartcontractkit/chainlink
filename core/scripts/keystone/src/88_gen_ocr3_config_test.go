package src

import (
	"errors"
	"testing"

	"github.com/gkampitakis/go-snaps/match"
	"github.com/gkampitakis/go-snaps/snaps"
)

func TestGenerateOCR3Config(t *testing.T) {
	// Generate OCR3 config
	config := generateOCR3Config("./testdata/SampleConfig.json", 11155111, "./testdata/PublicKeys.json")

	matchOffchainConfig := match.Custom("OffchainConfig", func(s any) (any, error) {
		// coerce the value to a string
		s, ok := s.(string)
		if !ok {
			return nil, errors.New("offchain config is not a string")
		}

		// if the string is not empty
		if s == "" {
			return nil, errors.New("offchain config is empty")
		}

		return "<nonemptyvalue>", nil
	})

	snaps.MatchJSON(t, config, matchOffchainConfig)
}
