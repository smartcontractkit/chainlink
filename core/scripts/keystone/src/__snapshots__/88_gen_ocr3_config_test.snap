
[TestGenerateOCR3Config - 1]
{
 "F": 1,
 "OffchainConfig": "<nonemptyvalue>",
 "OffchainConfigVersion": 30,
 "OnchainConfig": "0x",
 "Signers": [
  "0xA2402db8E549f094EA31e1C0EDd77623F4cA5b12",
  "0x4af19c802B244D1d085492C3946391C965E10519",
  "0x61925685d2B80b121537341d063c4E57B2F9323c",
  "0xFd97eFD53FC20acc098Fcd746C04d8d7540D97E0",
  "0xA0b67Dc5345a71D02B396147AE2cb75dDA63CbE9"
 ],
 "Transmitters": [
  "0xF4e7e516146c8567F8E8be0ED1f1A92798628d35",
  "0x8B60FDcc9CAC8ea476b31d17011CB204471431d9",
  "0x6620F516F29979B214e2451498a057FDd3a0A85d",
  "0xFeB61E22FCf4F9740c9D96b05199F195bd61A7c2",
  "0x882Fd04D78A7e7D386Dd5b550f19479E5494B0B2"
 ]
}
---
